src/kernel/thread.c

```diff
@@ -137,6 +141,11 @@ void doReplyTransfer(tcb_t *sender, tcb_t *receiver, cte_t *slot, bool_t grant)
     reply_remove(reply, receiver);
     assert(thread_state_get_replyObject(receiver->tcbState) == REPLY_REF(0));
     assert(reply->replyTCB == NULL);
+
+    if (sc_sporadic(receiver->tcbSchedContext) && sc_active(receiver->tcbSchedContext)
+        && receiver->tcbSchedContext != NODE_STATE(ksCurSC)) {
```

Comment:
I think this line would need to have `NODE_STATE_ON_CORE(ksCurSC, receiver
->tcbSchedContext->scCore)` here to cover the scenario where the reply object is blocked on by a thread on a different core, and a different thread on the same core is currently executing with the SC held in that core's ksCurSC?

Reply:
This was only trying to exclude the case that the current SC was donated back to the receiving thread. If the SC on the thread isn't the current SC for the current core, it can't be the current SC on any core because it must have been bound to the thread that was blocked on the reply object.

Reply:
Ah, I see your point though. In which case your suggested change would be appropriate.